plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSource.java

```diff
@@ -169,7 +200,103 @@ static TrinoException handleException(ParquetDataSourceId dataSourceId, Exceptio
         return new TrinoException(HIVE_CURSOR_ERROR, format("Failed to read Parquet file: %s", dataSourceId), exception);
     }
 
-    private static Block getRowIndexColumn(long baseIndex, int size)
+    private static boolean isColumnAdaptationRequired(List<ColumnAdaptation> columnAdaptations)
+    {
+        boolean isColumnAdaptationRequired = false;
+        // If no synthetic columns are added and the source columns are in order, no adaptations are required
+        for (int columnChannel = 0; columnChannel < columnAdaptations.size(); columnChannel++) {
+            ColumnAdaptation column = columnAdaptations.get(columnChannel);
+            if (column instanceof SourceColumn) {
+                int delegateChannel = ((SourceColumn) column).getSourceChannel();
+                if (columnChannel != delegateChannel) {
+                    isColumnAdaptationRequired = true;
+                    break;
+                }
+            }
+            else {
+                isColumnAdaptationRequired = true;
+                break;
+            }
+        }
+        return isColumnAdaptationRequired;
+    }
+
+    private interface ColumnAdaptation
+    {
+        Block block(Page sourcePage, long startRowId);
```

Comment:
"block" is a noun and method names should be verbs. `getBlock` would probably work.